pkg/phlaredb/block_querier.go

```diff
@@ -840,26 +840,8 @@ func MergeProfilesStacktraces(ctx context.Context, stream *connect.BidiStream[in
 		for _, querier := range queriers {
 			querier := querier
 			g.Go(util.RecoverPanic(func() error {
-
-				iters, err := querier.SelectMatchingProfiles(ctx, request)
-				if err != nil {
-					return err
-				}
-				defer func() {
-					iters.Close()
-				}()
-
-				profiles, err := iter.Slice(iters)
-				if err != nil {
-					return err
-				}
-
-				if len(profiles) == 0 {
-					return nil
-				}
-
 				// TODO(simonswine): Split profiles per row group and run the MergeByStacktraces in parallel.
-				merge, err := querier.MergeByStacktraces(ctx, iter.NewSliceIterator(querier.Sort(profiles)))
```

Comment:
Basically sums up the this PR. No need to order by timestamp than series via slices then back to series then timestamp via querier.Sort.